openapi-cli/src/main/java/io/ballerina/openapi/generators/service/RequestBodyGenerator.java

```diff
@@ -76,30 +79,26 @@
      */
     public RequiredParameterNode createNodeForRequestBody(Components components, RequestBody requestBody)
             throws BallerinaOpenApiException {
-        Token comma = createToken(SyntaxKind.COMMA_TOKEN);
         List<Node> literals = new ArrayList<>();
-        MappingConstructorExpressionNode annotValue;
+        MappingConstructorExpressionNode annotValue = null;
         TypeDescriptorNode typeName;
-        if (requestBody.getContent().entrySet().size() > 1) {
-            IdentifierToken mediaType = createIdentifierToken(MEDIA_TYPE_KEYWORD);
-            // Filter same data type
-            HashSet<Map.Entry<String, MediaType>> equalDataType = filterMediaTypes(requestBody);
-            if (!equalDataType.isEmpty()) {
-                typeName = getNodeForPayloadType(components, equalDataType.iterator().next());
-                SeparatedNodeList<MappingFieldNode> fields = fillRequestAnnotationValues(comma, literals, mediaType,
-                        equalDataType);
+        // Filter same data type
+        HashSet<Map.Entry<String, MediaType>> equalDataType = filterMediaTypes(requestBody);
+        if (!equalDataType.isEmpty()) {
+            typeName = getNodeForPayloadType(components, equalDataType.iterator().next());
+            SeparatedNodeList<MappingFieldNode> fields = fillRequestAnnotationValues(literals, equalDataType);
+            annotValue = NodeFactory.createMappingConstructorExpressionNode(
+                    createToken(SyntaxKind.OPEN_BRACE_TOKEN), fields, createToken(SyntaxKind.CLOSE_BRACE_TOKEN));
+        } else {
+            Iterator<Map.Entry<String, MediaType>> content = requestBody.getContent().entrySet().iterator();
+            Map.Entry<String, MediaType> mime = content.next();
+            equalDataType.add(mime);
+            typeName = getNodeForPayloadType(components, mime);
+            if (mime.getKey().equals(APPLICATION_URL_ENCODE)) {
+                SeparatedNodeList<MappingFieldNode> fields = fillRequestAnnotationValues(literals, equalDataType);
```

Comment:
Is there any a particular reason not to use streaming APIs, instead of using this iterator?

Reply:
Here we didn't iterate all the elements in the Map set. Our intention is to pick the first element from mediaType. To achieve this  I used the first iteration there. since we are unable to predict the key for picking the value directly taken. 🙂

Reply:
I'm not sure why we don't need to consider the other elements apart from the first element. Also, is this a ordered map (i.e. `LinkedHashMap`)? Otherwise we cant guarantee on the order right?

Reply:
This is the scenario we are going to address here 
```openapi
paths:
  /pets:
    post:
      summary: Add a new pet
      requestBody:
        description: Optional description in *Markdown*
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Pet'
          application/xml:
            schema:
              $ref: '#/components/schemas/PetList'
          text/plain:
            schema:
              type: string
```
Here request body has multiple mediaType and those media types map to different payload types. In ballerina, currently, we have no way to support these multiple behaviours, Therefore we go with the first media type and its payload type for the resource function request body details.  That is why we pick the first element here. 🙂 
When come to access it in openapi spec parser `requestBody.getContent()` give output as [`Content`](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-models/src/main/java/io/swagger/v3/oas/models/media/Content.java) type object its extend from `LinkedHashMap<String, MediaType>` . Yeah it is ordered map. 🙂

Reply:
Hmmm it makes sense.. In that case shouldn't we have at-least a warning, mentioning that we ignoring the other types?

Reply:
Yeah it would be great to have warning , I will add a warning there :)

Reply:
I will address warning via this issue https://github.com/ballerina-platform/openapi-tools/issues/942 :)